### ISSUE TITLE
[4.0] TypeError: this.querySelector(...).open is not a functionjoomla-field-media.js:244:45

### ISSUE THREAD

**Issue by uglyeoin** [joomla/joomla-cms#26711]
### Steps to reproduce the issue
Create a custom field for repeatable (subform).
Create a media field within repeatable (subform).
Click on media as if to upload an image

Here is a video explanation:
https://youtu.be/2zwh2Tvayb4

If you click save you can then update the item.

### Expected result
Can upload media

### Actual result
Nothing happens.  Console shows this JS error.
TypeError: this.querySelector(...).open is not a functionjoomla-field-media.js:244:45

**Comment by sebenns** [comment]
I can confirm this and I already can guess why that is not working, but I have neither the time to rewrite that nor do I know, what problems could occur after a rewrite.

Okay: Try this -> 

1. Create a custom field for repeatable in the com_fields component and add media field within that (form fields down below).
2. Create a 2nd custom field for media uploads - just for testing purpose.
2. Edit any user and open the custom fields tab.
3. On the one side you have your repeatable and on the other your working media upload.
4. Add a media field in the repeatable by pressing the plus button.
5. The *dynamic* generated media field wont open a modal.
6. Save that page now, and try it again. It will work now. 

For those who just want to see that in action, I recorded a small video:
http://sebenns.com/src/VVw4ulL6Iv.mp4

Tl;dr: In the way it is actually implemented it won't work with dynamic rendering.
Modal gets rendered in [media.php](https://github.com/joomla/joomla-cms/blob/staging/layouts/joomla/form/field/media.php "media.php") (124-137).





<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26711">issues.joomla.org/tracker/joomla-cms/26711</a>.</sub>

**Comment by Quy** [comment]
Please test #30455

**Comment by HRIT-Florian** [comment]
Problem still exists again:

![grafik](https://user-images.githubusercontent.com/12803406/109434430-8ccbcb00-7a15-11eb-8a2b-113f5ac41c45.png)
Steps to reproduze, see here: #28953 (same here!)

@Quy please reopen issue!
